File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/Rect.java
package io.appium.espressoserver.lib.helpers;

/**
 * Copies parts of the Android Rect
 *
 * This is copied so that we can run in Java context
 */
public class Rect {

    private final int left;
    private final int top;
    private final int right;
    private final int bottom;

    public Rect(int left, int top, int right, int bottom) {
        this.left = left;
        this.top = top;
        this.right = right;
        this.bottom = bottom;
    }

    public boolean contains(int x, int y) {
        return left < right && top < bottom  // check for empty first
                && x >= left && x < right && y >= top && y < bottom;
    }

    public String toString() {
        StringBuilder sb = new StringBuilder(32);
        sb.append("Rect("); sb.append(left); sb.append(", ");
        sb.append(top); sb.append(" - "); sb.append(right);
        sb.append(", "); sb.append(bottom); sb.append(")");
        return sb.toString();
    }

    public String toShortString() {
        StringBuilder sb = new StringBuilder(32);
        sb.setLength(0);
        sb.append('['); sb.append(left); sb.append(',');
        sb.append(top); sb.append("]["); sb.append(right);
        sb.append(','); sb.append(bottom); sb.append(']');
        return sb.toString();
    }
}

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/Rect.kt
package io.appium.espressoserver.lib.helpers

/**
 * Copies parts of the Android Rect
 *
 * This is copied so that we can run in Java context
 */
class Rect(private val left: Int, private val top: Int, private val right: Int, private val bottom: Int) {

    fun contains(x: Int, y: Int): Boolean {
        return (left < right && top < bottom  // check for empty first
                && x >= left && x < right && y >= top && y < bottom)
    }

    override fun toString(): String {
        return "Rect($left, $top, $right, $bottom)"
    }

    fun toShortString(): String {
        return "Rect[$left, $top, $right, $bottom]"
    }
}
